File: apps/web/src/lib/types/open-house.ts
export type OpenHouseStatus = "scheduled" | "cancelled" | "completed";

export interface OpenHouse {
  id: string;
  listing_id: string;
  starts_at: string;
  ends_at: string;
  capacity: number | null;
  notes: string | null;
  status: OpenHouseStatus;
  cancelled_reason: string | null;
  created_at: string;
  updated_at: string;
}

export interface OpenHouseCreate {
  starts_at: string;
  ends_at: string;
  capacity?: number | null;
  notes?: string | null;
}

export interface OpenHouseUpdate {
  starts_at?: string | null;
  ends_at?: string | null;
  capacity?: number | null;
  notes?: string | null;
}

export interface OpenHouseCancelRequest {
  reason?: string | null;
}

export interface OpenHouseListResponse {
  items: OpenHouse[];
  total: number;
}

export interface OpenHouseListParams {
  include_past?: boolean;
}

File: apps/api/schemas/open_house.py
"""Open house request/response schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from apps.api.models.open_house_enums import OpenHouseStatus


class OpenHouseCreate(BaseModel):
    starts_at: datetime
    ends_at: datetime
    capacity: int | None = Field(default=None, ge=1)
    notes: str | None = None


class OpenHouseUpdate(BaseModel):
    starts_at: datetime | None = None
    ends_at: datetime | None = None
    capacity: int | None = Field(default=None, ge=1)
    notes: str | None = None


class OpenHouseResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    listing_id: uuid.UUID
    property_id: uuid.UUID
    tenant_id: uuid.UUID
    starts_at: datetime
    ends_at: datetime
    capacity: int | None = None
    notes: str | None = None
    status: OpenHouseStatus
    created_by_user_id: uuid.UUID | None = None
    created_at: datetime
    updated_at: datetime


class OpenHouseListResponse(BaseModel):
    items: list[OpenHouseResponse]
    total: int


class OpenHouseCancelRequest(BaseModel):
    reason: str | None = None
